### ISSUE TITLE
Terraform v0.12-beta1 input variables in .tf files in subfolders are not being detected

### ISSUE THREAD

**Issue by wesmartin17** [hashicorp/terraform#20772]
Running terraform plan and apply fail to detect input variables that have been declared in a sub-folder.  This is not a problem in 0.11.  

### Terraform Version
Terraform v0.12.0-beta1
+ provider.aws v1.60.0-dev20190216h00-dev
+ provider.local v1.2.0
+ provider.random v2.1.0

Thanks

**Comment by apparentlymart** [comment]
Hi @wesmartin17!

Can you share a little more detail about exactly what you were doing in Terraform v0.11 that is no longer working? Auto-detection of `.tfvars` files (the `terraform.tfvars` and `*.auto.tfvars` files) was always supposed to work only in the current directory, so I'm curious to understand exactly what behavior you've been seeing in v0.11.

**Comment by wesmartin17** [comment]
Hi @apparentlymart thanks for getting back to me.

I previously had my project set up as: 

```
|--main.tf
|--variables.tf
```

with variables.tf containing input variables that have default values.  However the file was getting very large so I split it into two and made a folder called variables as such: 

```
|--main.tf
|--variables
    |--foo-variables.tf
    |--bar-variables.tf
```

But when I do terraform plan, I get 
     
```
Error: Reference to undeclared input variable

  on main.tf line 57, in provider "aws":
   2:   region = "${var.foo_bar}"

An input variable with the name "foo_bar" has not been declared. This
variable can be declared with a variable "foo_bar" {} block.
```

I have confirmed that my structure works in 0.11

**Comment by apparentlymart** [comment]
Hi @wesmartin17! Thanks for following up.

I'm very confused as to why what you did there is working in v0.11, since Terraform looks for `.tf` files only in a single directory unless there are `module` blocks in the root module. I assume there must be something else going on here that was making this work before, but I don't know what it was based on the information you've shared so far. Unfortunately I'm also not sure what more information to ask for, since this behavior you're seeing is so unexpected I don't really know where to look for it.

Could you perhaps do a run with Terraform v0.11 against this configuration with the environment variable `TF_LOG=trace` set, and then share the verbose trace log here as a [gist](https://gist.github.com/)? Then I might be able to understand from the logs how Terraform is finding the files in the subdirectory.

With all of that said: what you were doing in v0.11 was never intended to work, so I suspect it was only working due to a bug in v0.11, which we've now "fixed" for v0.12. I know that doesn't really help with your configuration as written, but I suspect we're going to have to require you to put it back how it used to be (all `.tf` files in the same directory) for v0.12 because changing the behavior now is very likely to have surprising consequences for existing users who were not depending on this apparent undocumented behavior.

**Comment by wesmartin17** [comment]
Ah my bad, I thought I recalled it being a valid structure and had asked my co-worker who still has 0.11 installed to test it for me.  I think he only ran plan so that might have been it.  I'll just continue working with a single level directory.  Thanks for helping me anyway!
